### PR TITLE
Fix generic Cache class

### PR DESCRIPTION
**Pull request by vadymmarkov**
1) After Swift 3.1 migration `Cache<T>` was slightly broken in a way that it was possible to add any `Cachable` types to it, so in fact it behaved as a `HybridCache`. 
2) Here I've also done some internal refactoring that shouldn't affect the public API.

### CODE REVIEW

**Comment by onmyway133** [comment]
@vadymmarkov can you elaborate a bit on the major idea of `internal refactoring`?

**Comment by vadymmarkov** [comment]
@onmyway133 I moved the code around to basically make this fix happen 😄

File: Source/Shared/BasicHybridCache.swift [hunk 1]
```diff
@@ -1,4 +1,8 @@
-import Foundation
+#if os(macOS)
+  import AppKit
```
Comment (zenangst) [inline Source/Shared/BasicHybridCache.swift]: AppKit? 😁

**Comment by zenangst** [comment]
Tests are failing on Travis 🤔

**Comment by zenangst** [comment]
@vadymmarkov I'm investigating the Travis configuration :)

**Comment by zenangst** [comment]
@vadymmarkov yay, changing the OS X image worked, feel free to merge 😎

**Comment by vadymmarkov** [comment]
@zenangst Great 👍

**Pull request merged**